Sources/HTTP/HTTPServer.swift

```diff
@@ -28,31 +31,72 @@ public class HTTPServer {
     /// To process incoming requests
     public let handler: HTTPRequestHandler
 
-    private let server = PoCSocketSimpleServer()
+    /// Stop the server
+    public func stop() {
+        server?.close()
+    }
+
+    /// The port number the server is listening on
+    public var port: Int {
+        return options.port
+    }
+
+    /// Reference to the server for closing
+    private var server: TCPServer?
+
+    /// The number of current connections
+    public var connectionCount: Int
 
     /// Create an instance of the server. This needs to be followed with a call to `start(port:handler:)`
     public init(with newOptions: Options, requestHandler: @escaping HTTPRequestHandler) {
         options = newOptions
         handler = requestHandler
+        connectionCount = 0
     }
 
     /// Start the HTTP server on the given `port` number, using a `HTTPRequestHandler` to process incoming requests.
     public func start() throws {
-        try server.start(port: options.port, handler: handler)
+        let socket = try TCPSocket(isNonBlocking: true, shouldReuseAddress: true)
+        var eventLoops: [DispatchQueue] = []
+        for i in 1...8 {
+            eventLoops.append(.init(label: "org.swift.server.eventLoop.\(i)"))
+        }
+        let server = TCPServer(socket: socket, eventLoops: eventLoops)
+        print("Starting on http://localhost:\(options.port)/")
+        try server.start(hostname: "0.0.0.0", port: UInt16(options.port), backlog: 4096) { client in
```

Comment:
I mean you use a (unreasonably?) huge backlog, but this seems to use a single accept queue. Since accept can take a moment (TCP handshake and all that crap, let alone a TLS driver ;-) ) and is a blocking call, I wonder whether this is viable.

Reply:
we use `O_NONBLOCK` so the accept [should be nonblocking](http://pubs.opengroup.org/onlinepubs/9699919799/functions/accept.html).

Reply:
This is of course completely right. 🤦‍♀️ :-)